gradle-plugin-testing/src/main/java/com/palantir/gradle/plugintesting/PluginTestingPlugin.java

```diff
@@ -158,9 +161,40 @@ public void execute(Task _task) {
             });
         });
 
+        addTestClassesDiscoveryTasks(project);
+
         addGradlePluginForTestingConfigurations(project);
     }
 
+    private static void addTestClassesDiscoveryTasks(Project project) {
+        project.getTasks().register("discoverGradlePluginTests", DiscoveryTestClassesTask.class, task -> {
+            task.getTestClassType().set("java");
+            task.getExtraArguments()
+                    .addAll(List.of(
+                            "withAnnotations", "--include", "com.palantir.gradle.testing.junit.GradlePluginTests"));
+        });
+
+        project.getTasks().register("discoverGroovyTestClassesToMigrate", DiscoveryTestClassesTask.class, task -> {
```

Comment:
perhaps let's call it `discoverNebulaTestClassesToMigrate` ?

Reply:
I think we might want to migrate spock test classes as well in the future - eg. Specification ? wdyt ?